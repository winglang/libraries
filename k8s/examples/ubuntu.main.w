bring "cdk8s-plus-27" as k8s;

let deployment = new k8s.Deployment(replicas: 3);
deployment.addContainer(image: "ubuntu");
