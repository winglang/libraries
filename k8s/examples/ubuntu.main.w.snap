apiVersion: apps/v1
kind: Deployment
metadata:
  name: deployment-c87ec702
spec:
  minReadySeconds: 0
  progressDeadlineSeconds: 600
  replicas: 3
  selector:
    matchLabels:
      cdk8s.io/metadata.addr: Deployment-c8c1f37c
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      labels:
        cdk8s.io/metadata.addr: Deployment-c8c1f37c
    spec:
      automountServiceAccountToken: false
      containers:
        - image: ubuntu
          imagePullPolicy: Always
          name: main
          resources:
            limits:
              cpu: 1500m
              memory: 2048Mi
            requests:
              cpu: 1000m
              memory: 512Mi
          securityContext:
            allowPrivilegeEscalation: false
            privileged: false
            readOnlyRootFilesystem: true
            runAsNonRoot: true
      dnsPolicy: ClusterFirst
      hostNetwork: false
      restartPolicy: Always
      securityContext:
        fsGroupChangePolicy: Always
        runAsNonRoot: true
      setHostnameAsFQDN: false
      terminationGracePeriodSeconds: 30
